src/App.tsx

```diff
@@ -1,24 +1,244 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react';
+import classNames from 'classnames';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  patchTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { SortType } from './types/SortType';
+import { Error } from './components/Error/Error';
 
-const USER_ID = 0;
+const USER_ID = 11050;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sort, setSort] = useState(SortType.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const completedTodo = todos.filter(todo => todo.completed === true);
+  const activeTodo = todos.length - completedTodo.length;
```

Comment:
apply `useMemo` here to avoid recalculation on each render

Reply:
not fixed